data/transactions/verify/txn.go

```diff
@@ -218,10 +235,10 @@ func stxnVerifyCore(s *transactions.SignedTxn, txnIdx int, groupCtx *GroupContex
 			return nil
 		}
 
-		return errors.New("signedtxn has no sig")
+		return signatureErrorf("signedtxn has no sig")
```

Comment:
A proposal : This is the _only_ interesting error that you need to notice and allow through.  We want submitters to be able to elide transactions, but I don't see a good reason to have a bunch of code that allows all the various kinds of signatures to fail.

If that's the case, we can remove all the wrapper signaturesError stuff, and just make this a named error, and use Error.Is to allow it.